### PR TITLE
feat: connecting the new Telegraf page to the edit config page

### LINKED ISSUE #1556: Telegraf Refresh: Create Config File
GIVEN a user has selected a bucket and plug-in, during the Add Plugin to New Config (Create a Telegraf Configuration) flow OR the Create a Configuration from the Telegraf Page flow,
WHEN the user selects the clicks the CTA to continue,
THEN the user is taken to the Create a Telegraf Configuration page, with a link to the selected plugin which opens the GitHub README documentation for the that plug-in AND
 opens an editor with the new input plugin config for the selected plugin and InfluxDB output config (including the selected output bucket) copied into the editor buffer, and the cursor at the top of the file.

https://www.figma.com/file/hMUcxA5RdoEhdJLvClWWB7/Agent-Configuration?node-id=263%3A0

### LINKED ISSUE #1745: Telegraf Refresh: Edit Config File
GIVEN a user has selected a bucket, configuration file, and plug-in, during the Add Plugin to Existing Config flow,
WHEN the user selects the clicks the CTA to continue,
THEN the user is taken to the `Edit this Telegraf Configuration` page,
which opens an editor with the new input plugin config for the selected plugin, copied into the editor buffer, and the cursor at the top of the copied portion, which will be appended to the bottom of the existing file.

https://www.figma.com/file/hMUcxA5RdoEhdJLvClWWB7/Agent-Configuration?node-id=263%3A54

### PR DESCRIPTION
**Pull request by zoesteinkamp**
Closes #1745 
Closes #1556 

https://user-images.githubusercontent.com/6667389/133513370-a1f32180-52b2-4abc-a973-806edbb8906f.mov

### CODE REVIEW

**Comment by TCL735** [comment]
Does this go through the verification process twice? Once from re-using the same pattern as the plugins' flow (which includes verification). And then a repeated verification from the `CollectorsStepSwitcher`?

**Comment by TCL735** [comment]
@zoesteinkamp - a video of this working locally would be great. Here is the idpe commit hash that will allow the backend to save properly: bb04995f956ff22a35ce97f0d466ddfc8ffc6549

You can also wait until idpe has fully deployed the latest merge in master and just update remocal with the latest.

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 59 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=2582)

File: src/dataLoaders/components/collectorsWizard/CollectorsStepSwitcher.tsx
Comment (hoorayimhelping) [inline src/dataLoaders/components/collectorsWizard/CollectorsStepSwitcher.tsx:39]: there's already a condition for

```ts
if (isFlagEnabled('telegrafUiRefresh')) {
...
```
it probably makes more sense to use that existing conditional statement

File: src/dataLoaders/components/collectorsWizard/select/SelectCollectorsStep2.tsx
Comment (hoorayimhelping) [inline src/dataLoaders/components/collectorsWizard/select/SelectCollectorsStep2.tsx:47]: this `<>` looks unnecessary since `<Form>` is the top level component returned

File: src/dataLoaders/components/collectorsWizard/select/StreamingSelector2.tsx
Comment (hoorayimhelping) [inline src/dataLoaders/components/collectorsWizard/select/StreamingSelector2.tsx:24]:
```suggestion
import {TelegrafPlugin as DataLoaderTelegrafPlugin} from 'src/types/dataLoaders'
```

`TelegrafPlugin2` is a little bit too vague I think

File: src/dataLoaders/components/collectorsWizard/verify/PluginConfigSwitcher.tsx
Comment (hoorayimhelping) [inline src/dataLoaders/components/collectorsWizard/verify/PluginConfigSwitcher.tsx:28]: should this be a spinner container?

File: src/dataLoaders/components/collectorsWizard/verify/PluginConfigSwitcher.tsx
Comment (hoorayimhelping) [inline src/dataLoaders/components/collectorsWizard/verify/PluginConfigSwitcher.tsx:78]: if there's no clear organization scheme, alphabetization is always a good one to use

**Review by hoorayimhelping — COMMENTED** [review]
Nice work. I have a couple of minor questions, not sure if they should block this. Look them over and let me know

File: src/dataLoaders/components/collectorsWizard/CollectorsStepSwitcher.tsx
Comment (zoesteinkamp) [reply src/dataLoaders/components/collectorsWizard/CollectorsStepSwitcher.tsx:39]: good point, went ahead and condensed these

File: src/dataLoaders/components/collectorsWizard/select/SelectCollectorsStep2.tsx
Comment (zoesteinkamp) [reply src/dataLoaders/components/collectorsWizard/select/SelectCollectorsStep2.tsx:47]: 100% not needed, removed

File: src/dataLoaders/components/collectorsWizard/select/StreamingSelector2.tsx
Comment (zoesteinkamp) [reply src/dataLoaders/components/collectorsWizard/select/StreamingSelector2.tsx:24]: good catch! not a good name indeed

File: src/dataLoaders/components/collectorsWizard/verify/PluginConfigSwitcher.tsx
Comment (zoesteinkamp) [reply src/dataLoaders/components/collectorsWizard/verify/PluginConfigSwitcher.tsx:28]: This is incase the next page takes some time to load, its also being used in the PluginCreateConfigurationWizard.tsx that timmy wrote. if we remove it here we should also remove it there and vise versa

File: src/dataLoaders/components/collectorsWizard/verify/PluginConfigSwitcher.tsx
Comment (zoesteinkamp) [reply src/dataLoaders/components/collectorsWizard/verify/PluginConfigSwitcher.tsx:78]: good deal

**Review by hoorayimhelping — APPROVED** [review]

**Pull request merged**
